.github/workflows/diff_report.yml

```diff
@@ -0,0 +1,150 @@
+      
+    steps:
+    
+     - name: Issue comment action
+       if: github.event_name == 'issue_comment'
+       run: |
+        echo "${{github.event.issue.body}}" > text
+        echo "${{github.event.issue.user.login}}" > user
+        wget -q ${{github.event.issue.pull_request.url}} -O info.json
+        jq .head.ref info.json > branch
+       
+     - name: PR action
+       if: github.event_name == 'pull_request'
+       run: |
+        echo "${{github.event.pull_request.body}}" > text
+        echo ${{github.event.pull_request.user.login}} > user
+        echo ${{github.event.pull_request.head.ref}} > branch
+        
+     - name: Parse body
+       id: parse
+       run: |
+        grep "^Diff Regression projects:" text > temp || echo "" > temp
+        sed 's/Diff Regression projects: //' temp > proj
+        echo ::set-output name=proj_link::$(cat proj)
+        grep "^Diff Regression config:" text > temp || echo "" > temp
+        sed 's/Diff Regression config: //' temp > config
+        echo ::set-output name=config_link::$(cat config)
+        
+     - name: Set branch and head_label
+       id: branch
+       run: |
+        echo ::set-output name=user::$(cat user)
+        echo ::set-output name=ref::$(cat branch)
+      
+     - name: info
+       run: |
+        cat user
+        cat branch
+        echo ${{steps.branch.outputs.ref}}
+        echo ${{steps.branch.outputs.user}}:
+     
+  make_report:
+    runs-on: ubuntu-latest
+    needs: parse_body
+    if: needs.parse_body.outputs.proj_link != '' && needs.parse_body.outputs.config_link != ''
+    steps: 
+       
+     - name: Download files
+       run: |
+        wget -q ${{needs.parse_body.outputs.proj_link}} -O proj.properties
+        wget -q ${{needs.parse_body.outputs.config_link}} -O config.xml
+        
+     - name: Download checkstyle
+       uses: actions/checkout@v2
+       with:
+        repository: ${{needs.parse_body.outputs.user}}/checkstyle
+        ref: master
+        path: checkstyle
+        fetch-depth: 0
+        
+     - name: Check branches
+       run: |
+         cd checkstyle
+         git branch -r
+         cd ../
+     
+     - name: Download contribution
+       uses: actions/checkout@v2
+       with:
+        repository: checkstyle/contribution
+        path: contribution
+         
+     - name: Print branch
+       run: |
+         echo ${{needs.parse_body.outputs.branch}}
+         
+     - name: Relocate .xml and .properties 
+       run: |
+         mv config.xml ./contribution/checkstyle-tester/
+         mv proj.properties ./contribution/checkstyle-tester/
+         
+     - name: Generate report
+       run: |
+         bash
+         REF="origin/${{needs.parse_body.outputs.branch}}"
+         cd contribution/checkstyle-tester
+         sudo apt-get install mercurial
+         sudo apt install groovy
+         groovy diff.groovy -r ../../checkstyle -b master -p $REF -c config.xml -l proj.properties
```

Comment:
Maybe thoughts for the future, to start a discussion:

What about handling PRs where only patch regression (no comparison to master)? This is for things like new checks.
What about handling PRs where base config is different from patch config? This is for things like where we split properties, change property types, add a new property, etc...

Reply:
all this can be done by special keywords in PR description.
and extra logic in this file, @strkkk , please try to make it in next PRs.

Reply:
@rnveach that is a good point. I think it can be done as separate issue(s).
I guess most of reports are for differences between master and branch for same config.

Reply:
@rnveach @romani 
I have created https://github.com/checkstyle/checkstyle/issues/8448 and https://github.com/checkstyle/checkstyle/issues/8447 for this